libraries/AP_Filesystem/AP_Filesystem_backend.h

```diff
@@ -22,6 +22,8 @@
 
 #include "AP_Filesystem_Available.h"
 
+#include <AP_InternalError/AP_InternalError.h>
```

Comment:
maybe add #if CONFIG_HAL_BOARD == HAL_BOARD_SITL there too to prevent unnecessary inclusion